.github/workflows/pr-check-links.yml

```diff
@@ -15,11 +15,26 @@ jobs:
           ref: ${{ github.event.pull_request.head.ref }}
           repository: ${{ github.event.pull_request.head.repo.full_name }}
 
-      - name: Check out main branch
-        run: git checkout main
+      - name: Set environment variables
+        id: set-env
+        run: |
+          # Extract the base commit ID where the feature branch diverged from main
+          base_commit=$(git merge-base origin/main ${{ github.event.pull_request.head.ref }})
+          echo "base_commit=$base_commit" >> $GITHUB_ENV
+
+          # Extract the head commit ID on the feature branch
+          head_commit=${{ github.event.pull_request.head.sha }}
+          echo "head_commit=$head_commit" >> $GITHUB_ENV
+
+      - name: Create baseline branch by reverting feature branch changes
+        run: |
+          # Create a copy of the feature branch
+          git checkout -b baseline-copy
```

Comment:
What's the reason for needing a copy of this branch?

Reply:
Basically, our main branch and feature branch can be presented like this:

```mermaid
gitGraph
   commit id: "Commit A"
   commit id: "Commit B"
   commit id: "Commit C" tag: "base commit"
   branch feature
   commit id: "Feature Commit 1"
   commit id: "Feature Commit 2"
   checkout main
   commit id: "Commit D"
```

I create a copy of my feature branch (`baseline-copy`) and then hard-reset it to the base commit to dump all links that exist in the repo at the time the feature branch was created:

```mermaid
gitGraph
   branch baseline-copy
   commit id: "Commit A"
   commit id: "Commit B"
   commit id: "Commit C" tag: "base commit"
```

In this way, I can compare the links added or altered in the feature branch with the links present in the main branch at the time the feature branch was created.